graylog2-web-interface/src/components/sidecars/configuration-forms/ConfigurationForm.tsx

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import PropTypes from 'prop-types';
+import React, { useState, useRef, useEffect } from 'react';
+import lodash from 'lodash';
+
+import history from 'util/History';
+import { ColorPickerPopover, FormSubmit, Select, SourceCodeEditor } from 'components/common';
+import { Button, Col, ControlLabel, FormControl, FormGroup, HelpBlock, Row, Input } from 'components/bootstrap';
+import Routes from 'routing/Routes';
+import ColorLabel from 'components/sidecars/common/ColorLabel';
+import { CollectorConfigurationsActions } from 'stores/sidecars/CollectorConfigurationsStore';
+import { CollectorsActions } from 'stores/sidecars/CollectorsStore';
+
+import SourceViewModal from './SourceViewModal';
+import ImportsViewModal from './ImportsViewModal';
+import ConfigurationTagsSelect from './ConfigurationTagsSelect';
+
+import type { Collector, Configuration, ConfigurationSidecarsResponse } from '../types';
+
+type Props = {
+  action: string,
+  configuration: Configuration,
+  configurationSidecars: ConfigurationSidecarsResponse,
+};
+
+const ConfigurationForm = ({
```

Comment:
There is a console error about ref being passed to this component 

`
react_devtools_backend.js:4026 Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of `SidecarNewConfigurationPage`.
    in ConfigurationForm (created by SidecarNewConfigurationPage)
    in div (created by Col)
    in Col (created by SidecarNewConfigurationPage)
    in div (created by Row)
    in Row (created by Row)
    in Row (created by SidecarNewConfigurationPage)
    in span (created by SidecarNewConfigurationPage)
    in DocumentTitle (created by SidecarNewConfigurationPage)
    in SidecarNewConfigurationPage
    in Suspense`